six/modules/c++/six.sidd/tests/test_geotiff.cpp

```diff
@@ -0,0 +1,168 @@
+/* =========================================================================
+* This file is part of six.sidd-c++
+* =========================================================================
+*
+* (C) Copyright 2004 - 2016, MDA Information Systems LLC
+*
+* six-c++ is free software; you can redistribute it and/or modify
+* it under the terms of the GNU Lesser General Public License as published by
+* the Free Software Foundation; either version 3 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public
+* License along with this program; If not,
+* see <http://www.gnu.org/licenses/>.
+*
+*/
+
+#include <iostream>
+#include <iterator>
+#include <import/six/sidd.h>
+#include "six/NITFWriteControl.h"
+#include "six/Types.h"
+
+namespace
+{
+static const std::string OUTPUT_NAME("GeoTIFFTest");
+static const size_t DATA_LENGTH = 100;
+static const size_t DATA_SIZE_IN_BYTES = DATA_LENGTH * sizeof(sys::Int16_T) / sizeof(six::UByte);
+static const size_t BYTES_PER_PIXEL = sizeof(sys::Int16_T);
+
+void generateData(mem::ScopedArray<sys::Int16_T>& data)
```

Comment:
Since you're not doing anything with the fact that this is a smart pointer, have the function just take in a raw pointer